### PR TITLE
Delete queued messages with button in preview

### PR DESCRIPTION
**Pull request by justanotherdot**
Ref #247.

Sorry this took a second to get up! Works been keeping me busy.

The placement and styling of the button can probably be configured from here; what's the general thoughts?

Here's what it looks like as is:

kapture 2018-06-18 at 13 43 47

### CODE REVIEW

**Comment by princemaple** [comment]
Hi @justanotherdot Thanks for the PR!

We don't normally modify server data with a get request. Even though this is dev only, could we do that with a POST request? An inline form that wraps the button should do it. What do you think?

**Comment by justanotherdot** [comment]
Yes, it is a little unprincipled. I don't mind either way; I can have the handler for the button produce a fetch and and redirect after the promise is resolved so as to choose a different method for the request.

File: lib/plug/templates/mailbox_viewer/index.html.eex [hunk 1]
```diff
@@ -98,6 +98,7 @@
                 <h6 class="list-group-item-heading">Empty mailbox...</h6>
               </div>
             <% else %>
+              <button class="btn btn-primary btn-block" onclick="location.href = '/clear'">Delete Message Queue</button>
```
Comment (princemaple) [inline lib/plug/templates/mailbox_viewer/index.html.eex]:
```html
<form method="POST" action="/clear">
  <button class="btn btn-primary btn-block" type="submit">Delete Message Queue</button>
</form>
```

File: lib/plug/mailbox_preview.ex [hunk 2]
```diff
@@ -46,6 +46,14 @@ if Code.ensure_loaded?(Plug) do
       |> send_resp(200, template(emails: emails, email: nil, conn: conn))
     end
 
+    get "/clear" do
```
Comment (princemaple) [inline lib/plug/mailbox_preview.ex]: `post "/clear"`

**Comment by justanotherdot** [comment]
> something like this should do? no need for js.

Definitely, just a super rough idea that popped off the top of my head. Thanks for the reminder about form method attr's 🙂

I'll doll that up now.

**Comment by justanotherdot** [comment]
@princemaple any thoughts on the layout for the button?

**Comment by princemaple** [comment]
I haven't tried to run this yet, what does it look like now?

**Comment by justanotherdot** [comment]
> I haven't tried to run this yet, what does it look like now?

Should be same as before, here's a new gif:

kapture 2018-06-18 at 16 26 01

The `margin-bottom` is to counteract the 1em `margin-bottom` being applied to `form`. I think that's possibly coming from bootstrap.

**Comment by justanotherdot** [comment]
> The margin-bottom is to counteract the 1em margin-bottom being applied to form. I think that's possibly coming from bootstrap.

to make things a little bit more tidy: shall I move that up into the inline CSS up top?

**Comment by princemaple** [comment]
That would be great, thanks. Thanks a lot!

**Pull request merged**